src/access/universal.test.ts

```diff
@@ -0,0 +1,889 @@
+    );
+
+    expect(getResourceInfoWithAcr).toHaveBeenCalledWith(
+      "https://some.pod/resource",
+      options
+    );
+  });
+
+  it("passes the provided fetch function to the WAC module", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    const mockedResourceWithAcl = addMockResourceAclTo(mockedResource);
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResourceWithAcl as any);
+    const wacSetAgentAccess = jest.spyOn(wacModule, "setAgentResourceAccess");
+    wacSetAgentAccess.mockResolvedValue(null);
+    const options = { fetch: () => new Promise(jest.fn()) as any };
+
+    await setAgentAccess(
+      "https://arbitrary.pod/resource",
+      "https://some.pod/profile#me",
+      { read: true },
+      options
+    );
+
+    expect(wacSetAgentAccess).toHaveBeenCalledWith(
+      expect.anything(),
+      "https://some.pod/profile#me",
+      { read: true },
+      options
+    );
+  });
+
+  it("returns null if the Resource has neither an ACR nor an ACL", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResource as any);
+
+    const access = await setAgentAccess(
+      "https://arbitrary.pod/resource",
+      "https://arbitrary.pod/profile#me",
+      { read: true }
+    );
+
+    expect(access).toBeNull();
+  });
+});
+
+describe("getGroupAccess", () => {
+  it("calls out to the well-tested ACP API for Resources with an ACR", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    const mockedAcr = mockAcrFor("https://arbitrary.pod/resource");
+    const mockedResourceWithAcr = addMockAcrTo(mockedResource, mockedAcr);
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResourceWithAcr);
+    jest.spyOn(acpModule, "internal_getGroupAccess");
+
+    await getGroupAccess(
+      "https://arbitrary.pod/resource",
+      "https://arbitrary.pod/groups#group"
+    );
+
+    expect(acpModule.internal_getGroupAccess).toHaveBeenCalledTimes(1);
+  });
+
+  it("calls out to the well-tested WAC API for Resources with an ACR", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    const mockedResourceWithAcl = addMockResourceAclTo(mockedResource);
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResourceWithAcl as any);
+    const wacGetGroupAccess = jest.spyOn(wacModule, "getGroupAccess");
```

Comment:
Seeing them side-by-side, this reminds me that the WAC functions should be prefixed with `internal`, right ? It's not necessary to do it in this PR though.

Reply:
Technically, yeah. I'll try to think of it after merging this.